AlpsSDK/ProximityHandler.swift

```diff
@@ -0,0 +1,144 @@
+//
+//  ProximityHandler.swift
+//  AlpsSDK
+//
+//  Created by Wen on 26.10.17.
+//  Copyright © 2017 Alps. All rights reserved.
+//
+
+import Foundation
+import CoreLocation
+import Alps
+
+protocol ProximityHandlerDelegate : class {
+    func didRangeBeacons(manager: ContextManager, beacons: [CLBeacon], knownBeacons: [IBeaconTriple])
+}
+
+final class ProximityHandler: ProximityHandlerDelegate {
+    var refreshTimer: Int = 60 * 1000 // timer is in milliseconds
+    lazy var beaconsDetected: [CLProximity: [IBeaconTriple]] = [:]
+    lazy var beaconsTriggered: [CLProximity: [String: ProximityEvent]] = [:]
+    
+    init() {
+        beaconsTriggered[CLProximity.unknown] = [String: ProximityEvent]()
+        beaconsTriggered[CLProximity.immediate] = [String: ProximityEvent]()
+        beaconsTriggered[CLProximity.far] = [String: ProximityEvent]()
+        beaconsTriggered[CLProximity.near] = [String: ProximityEvent]()
+    }
+    
+    func didRangeBeacons(manager: ContextManager, beacons: [CLBeacon], knownBeacons: [IBeaconTriple]) {
+        let unknownBeacons = {
```

Comment:
it can be just:
`unknownBeacons = beacons.filter {$0.proximity == CLProximity.unknown}`